### ISSUE TITLE
Advertising and data analytics

### ISSUE THREAD

**Issue by wseltzer** [w3c/strategy#81]
Standards for audience-data interchange, advertising, analytics? At W3C, it could build privacy in from the start. 

Noted while at a podcasting discussion. Will the data be/make the platform?

**Comment by liamquin** [comment]
Related: Accessibility - how do I as a Web site publisher make my Web page be accessible (or validate) when it contains fixed-pixel-size snippets whose content I don't control (images with text in them, coloured text in an iframe with !important in the styles, inline script-generated content, Flash...) and how can an advertiser make ads that are responsive and accessible in such a framework?

**Comment by liamquin** [comment]
Next steps after a call with IAB and Neilsen today -
. continue phone calls
. arrange an informal meeting - if we can get it hosted by iab canada in toronto (see below) that might be good politics; if not, probably NYC or San Jose. But it'd be good to make it "international" at least a little bit. If we have a full-blown workshop on technical changes to the Web platform to facilitate a better user experience and more effective monetization it should probably be in the continental EU (Brussels?) to give privacy-regulation-applies vibes.

An optimal outcome might be
* short term: an IG for visitor-based business models (i.e. ad-driven sites, help me out here :D :D )
* longer term: I suppose an IDPF-style merger with IAB, but anent that, maybe some joint technical work could be very productive all round. 

?joint work with webappsec?
?the Verifiable Claims (and Web payments in general) people are close to this space

On the call today we heard that there is significant interest in publishers (Web site owners) wanting/needing to be able to request accessible ads from the channels/distributors.

Specs like openrtb for real-tiime ad auctions - https://www.iab.com/guidelines/real-time-bidding-rtb-project/ - currently in JSON: web performance is an issue, EXI4JSON might help them, especially on WoT where EXI is already in use.

Analytics and deciding what's an ad impression - knowing when an element is rendered and viewable and "above the fold" (i.e. not scrolled below the visible region) is worth a lot of money to a lot of people. But it also matters to people with e.g. animated logos (stop the animation when it's not visible, to save battery/CPU).
Reporting pageviews through proxy servers has also always been an issue (that might be a thing to take jointly with IETF).

*

Some people outside Interactive Advertising Bureau (www.IAB.com) and Neilsen
Digital Analytics Association
. https://www.digitalanalyticsassociation.org/board-of-directors
 (note: Director Jeff Allan is from Adobe, who are also into Analytics;
 we should involve Adobe & of course Microsoft)

. interactive advertising Bureau of Canada, https://iabcanada.com/research/annual-internet-advertising-revenue-reports/
 I see they're having a multi-city event in October so there maybe a speaking opportunity.
 IAB Canada is based in Toronto. I note Amazon is a Member.

. Internet Advertising Bureau UK (abuk.net)
There's others, e.g. iabfrance.com, and Bundesverband Digitale Wirtschaft (BVDW) e.V. in Germany,
and also iabeurope.eu


Some players in the ad platform and ad reselling space
The two giants are google adsense and facebook;  google also sells ads as doubleclick, which they bought years ago.
https://www.adrecover.com/ - ads that aren't detected by adblock (In India??)
media.net (Yahoo/Bing)
Ad Maven (500 million ads served per day they say)
Infolinks, Chitika, the list goes on. I think it's too early to start covering the ad networks, that we should have a preparatory workshop first.

**Comment by ylafon** [comment]
See https://iabtechlab.com/ads-txt/

**Comment by liamquin** [comment]
Note, as of Nov 2017 we have an Improving Web Advertising business group, although so far we have only 3 or 4 organizations involved.

**Comment by Malvoz** [comment]
@liamquin 
> how do I as a Web site publisher make my Web page be accessible (or validate) when it contains fixed-pixel-size snippets whose content I don't control (images with text in them, coloured text in an iframe with !important in the styles, inline script-generated content, Flash...) ...

You may disallow inline scripts and styles using a [content security policy](https://www.w3.org/TR/CSP3/).

**Comment by liamquin** [comment]
On Fri, 2018-02-23 at 11:40 -0800, Malvoz wrote:
> @liamquin 
> > how do I as a Web site publisher make my Web page be accessible (or
> > validate) when it contains fixed-pixel-size snippets whose content
> > I don't control (images with text in them, coloured text in an
> > iframe with !important in the styles, inline script-generated
> > content, Flash...) ...
> 
> You may disallow inline scripts and styles using a [content security
> policy](https://www.w3.org/TR/CSP3/).

This doesn't guarantee accessibility, and would simply end up with
broken interactive ads on the publisher's page, which doesn't really
help anyone. In addition, the ad-injected iframe will point to the ad
distribution channel's servers for content, style and and scripts, so
i'm not sure CSP really helps.

What's (i _think_) wanted is metadata to give to the ad channel
distributor to say, "only accessible content on this page please".


-- 
Liam Quin, W3C, http://www.w3.org/People/Quin/
Staff contact for Verifiable Claims WG, SVG WG, XQuery WG
Improving Web Advertising: https://www.w3.org/community/web-adv/
Personal: awesome vintage art: http://www.fromoldbooks.org/

**Comment by Malvoz** [comment]
@liamquin 

Sorry, my comment added no real value to this discussion. 

> What's (i _think_) wanted is metadata to give to the ad channel
distributor to say, "only accessible content on this page please".

Are you also establishing advertiser guidelines to creating accessible ads?
Advertisers should follow the [Web Content Accessibility Guidelines](https://www.w3.org/TR/WCAG21/).

### Here are some essentials to an accessible ad:
### Accessible (and localized?) name:
The first step to an accessible ad is for users to be able to recognize that it is an ad.
[According to WCAG  2.0](https://www.w3.org/TR/WCAG20-TECHS/H64.html) `<iframe>`s need an [accessible name](https://www.w3.org/TR/accname-aam-1.1/), provided with the `title` attribute, should this have to be localized?

There is a great [blog post](http://davidmacd.com/blog/is-title-attribute-on-iframe-required-by-wcag.html) including a discussion on labelling `<iframe>`s .

### No keyboard traps:
https://www.w3.org/TR/WCAG21/#no-keyboard-trap

**Comment by liamquin** [comment]
On Sat, 2018-02-24 at 12:40 +0000, Malvoz wrote:
> @liamquin 
> 
> Sorry, my comment added no real value to this discussion. 
> 
> > What's (i _think_) wanted is metadata to give to the ad channel
> distributor to say, "only accessible content on this page please".
> 
> Are you also establishing advertiser guidelines to creating
> accessible ads?
> Advertisers should follow the [Web Content Accessibility Guidelines](
> https://www.w3.org/TR/WCAG21/).

WCAG currently has a cop-out clause for third-party content. But the
problem that i'm hoping the Improving Web Advertising BG will look at
in this area is this: when you load a Web page with ads, typically the
ads are supplied on the fly from an ad distribution channel (e.g.
Chitika, DoubleClick). Those ads are chosen based on keywords on the
Web page, from a pool supplied by advertisers and metadata about the
ads that the advertisers supply. Currently, there's no information
about whether the ad is accessible, so a page using third party ads
can't guarantee that the page is accessible. The advertisers not only
have no incentive today but no actual way e.g. to provide alt text.
They don't control the iframe, and the people who do, the channel
distributors and fulfillment networks, don't control the content of the
ads.

Another example is that e.g. Google text link ads (and ads from other
channels) are affected by the minimum font size set by the user in the
browser, but ad sizes are defined in terms of pixels, with overflow:
clip, which means i tend to see only the top half of a single line of
text and the links in the ads don't work.

So yes, WCAG helps, but it's more about organizations working together.

(It's possible Google's automatic ad placement and responsive ads will
work together in the future to mitigate the font size problem but i'm
not sure, and that's only one advertising company of many)

Thanks for replying. You could consider joining the advertising BG :)

Liam

-- 
Liam Quin, W3C, http://www.w3.org/People/Quin/
Staff contact for Verifiable Claims WG, SVG WG, XQuery WG
Improving Web Advertising: https://www.w3.org/community/web-adv/
Personal: awesome vintage art: http://www.fromoldbooks.org/

**Comment by Malvoz** [comment]
Could there be an Advertisement API to allow for communication between advertisers/ad channels and publishers?

**Comment by liamquin** [comment]
@Malvoz yes, we're thinking in terms of optional additional metadata that could be supplied by creatives and could be requested by publishers.

When you say that ad authors should never set a font size in pixels, i'd like to agree but today that's standard practice, e.g. for Google ads, and worse, the ad slots all have sizes measured in pixels, so if the user's minimum font size set in the browser is larger than the ad author expected, the text doesn't fit the ad. Some of that could be detected maybe by JavaScript in the ad fulfillment code. It might be that extending media queries would be a better approach, to reduce information leakage, i'm not sure.

On the question of an ad api, yes, there are such things, e.g. the open bid protocol. It'd be good to bring that work closer to the browser vendors, though.

**Comment by koalie** [comment]
https://hackaday.com/2020/03/30/solid-promises-a-new-approach-to-how-the-web-works/

SOLID Promises A New Approach To How The Web Works
by: Lewin Day 
March 30, 2020

Mentions Inrupt's solid, design for decentralized internet, data security, targeted advertising and business models, IoT.

> Solid is all about PODS – personal online data stores. [...] can be hosted at home, or with an external provider, but the point is that you choose and control where your data is hosted. PODS are intended to be portable. [...]instead of uploading data to remote services, the services are granted permission to access the data that lives in one of your PODS.

**Comment by wseltzer** [comment]
Closed and replaced by of #222
